src/modules/Elsa.MassTransit.AzureServiceBus/Features/AzureServiceBusFeature.cs

```diff
@@ -31,18 +53,81 @@ public override void Configure()
                 
                 configure.UsingAzureServiceBus((context, serviceBus) =>
                 {
+                    var options = context.GetRequiredService<IOptions<MassTransitWorkflowDispatcherOptions>>().Value;
+                    var instanceNameRetriever = context.GetRequiredService<IInstanceNameRetriever>();
+
                     if (ConnectionString != null) 
                         serviceBus.Host(ConnectionString);
-                    
                     serviceBus.UseServiceBusMessageScheduler();
                     ConfigureServiceBus?.Invoke(serviceBus);
+
+                    foreach (var consumer in shortLivedConsumers)
+                    {
+                        serviceBus.ReceiveEndpoint($"Elsa-{instanceNameRetriever.GetName()}-{consumer.Name}", configurator =>
+                        {
+                            configurator.AutoDeleteOnIdle = options.ShortTermQueueLifetime ?? TimeSpan.FromHours(1);
+                            configurator.ConcurrentMessageLimit = options.ConcurrentMessageLimit;
+                            configurator.ConfigureConsumer(context, consumer.ConsumerType);
+                        });
+                    }
+
                     serviceBus.ConfigureEndpoints(context, new KebabCaseEndpointNameFormatter("Elsa", false));
                 });
             };
         });
     }
+
+    private void RegisterConsumers(List<ConsumerTypeDefinition> consumers)
+    {
+        var subscriptionTopology = new List<MessageSubscriptionTopology>();
+        
+        foreach (var consumer in consumers)
+        {
+            foreach (var consumerInterface in consumer!.ConsumerType.GetInterfaces())
+            {
+                if (!consumerInterface.IsGenericType ||
+                    consumerInterface.GetGenericTypeDefinition() != typeof(IConsumer<>))
+                {
+                    continue;
+                }
+
+                var genericType = consumerInterface.GetGenericArguments()[0];
+                //While the name might show up in the Azure portal with a ~, the separator is actually an /
+                //see https://learn.microsoft.com/en-us/archive/blogs/servicebus/azure-service-bus-azure-resource-manager-and-this-character
+                var topicName = $"{genericType.Namespace.ToLower()}/{genericType.Name.ToLower()}";
+                    
+                subscriptionTopology.Add(new MessageSubscriptionTopology(topicName,
+                    consumer.Name ?? genericType.Name.ToLower(),
+                    consumer.IsShortLived));
+            }
+        }
+
+        Services.AddSingleton(new MessageTopologyProvider(subscriptionTopology));
+    }
+
+    /// <inheritdoc />
+    public override void Apply()
+    {
+        Services.Configure(AzureServiceBusOptions);
+        Services.AddScoped(ServiceBusAdministrationClientFactory);
+        Services.AddNotificationHandler<OrphanedSubscriptionRemover>();
```

Comment:
When naming event/notification handlers, we either use the form `HeartbeatTimedOutHandler`, `OrphanedSubscriptionHandler`, `RemoveOrphanedSubscription` or even `RemoveOrphanedSubscriptionHandler`.

The form `OrphanedSubscriptionRemover` is used more commonly for services that represent a thing, while the other forms are more commonly used for classes that represent a handler of an event.

You could have a service that is called `OrphanedSubscriptionRemover` that is then used by a handler - if we needed this functionality to be reusable, for example.